src/dropdown/dropdown.ts

```diff
@@ -324,16 +334,30 @@ export class NgbDropdown implements AfterContentInit, OnDestroy {
 
     if (key === Key.Tab) {
       if (event.target && this.isOpen() && this.autoClose) {
-        if (this.container === 'body' && this._anchor.anchorEl === event.target && !event.shiftKey) {
-          itemElements[0].focus();
-          event.preventDefault();
-        } else if (this.container === 'body' && event.shiftKey && position === 0) {
-          this._anchor.anchorEl.focus();
-          event.preventDefault();
+        if (this._anchor.anchorEl === event.target) {
+          if (this.container === 'body' && !event.shiftKey) {
+            this._renderer.setAttribute(this._menuElement.nativeElement, 'tabindex', '0');
+            this._menuElement.nativeElement.focus();
+            this._renderer.removeAttribute(this._menuElement.nativeElement, 'tabindex');
```

Comment:
We don't want the menu to be permanently focusable. It would make reachable with keyboard, which we clearly don't won't.

The trick here is to be able to _find_ the next focusable element in the menu, mostly without to specify it. Previous PR #3625 was wrongly assuming that focusable elements were only `ngbDropdownItem`. Simply have a look at the [form demo](https://ng-bootstrap.github.io/#/components/dropdown/examples#navbar)

To do that, I listen on the `keydown.Tab` from the menu anchor/toggle element. And when we are about to enter the menu, I temporarily and synchronously focus the menu, so that the browser will natively continue to forward the `Tab` **inside** the menu, hence focusing whatever focusable coming first.

Do you want me to comment it out in the code ?

Reply:
I see now, please do add a short comment